Add suspendable to figwheel system

featheredtoast opened the pull request:
closes #570

danielcompton commented on sidecar/src/figwheel_sidecar/system.clj:
```diff
@@ -130,7 +131,13 @@
+  (-actual [this] (:figwheel-server @system))
+  Suspendable
+  (suspend [c] c)
+  (resume [c old-c]
```
I think you want to check if it was suspended, and then assoc the values on, and otherwise start the component. See https://github.com/weavejester/ring-jetty-component/blob/87ac67ebc1d9cbc8fd340c880e5bc9a2fd8ac392/src/ring/component/jetty.clj#L25-L40 for a more complete suspendable implementation.

I don't think this will resume properly in all cases.

featheredtoast replied:
Thanks for the tip; I'll take another pass soon.
Just so I can understand, is the concern here if a resume is called before the component is started (or after it's stopped)?

danielcompton replied:
Yeah, that's right, ideally `resume` and `suspend` are idempotent, in the same way that `start` and `stop` are.

featheredtoast commented:
Just added :suspended? flag on suspend, and checks for it on resume. Otherwise, starts the component. Thanks @danielcompton for the references.

Renamed c and old-c to this and old-this to be more in line with how the components are named in figwheel.

bhauman commented:
looks great!
